### ISSUE TITLE
error naming inconsistencies

### ISSUE THREAD

**Issue by geemus** [excon/excon#186]
Some errors are `Excon::Errors::Foo` while others (most) are `Excon::Errors::FooError`. Should standardize (probably with aliasing/inheritance for backwards compatibility).

**Comment by starbelly** [comment]
@geemus For example `Excon::Errors::HTTPStatusError` ? Definitely redundant. Must normalize all the things! How do you feel about refactoring Excon::Errors into Excon::Error? Something like: 

    module Excon
      module Error
        def self.new(msg)
            Excon::Error::Standard.new(msg)
        end
        class Standard < StandardError; end
        class StubNotFound < Error::Standard; end
      end
      Excon::Errors = Excon::Error.dup
    end

    > Excon::Errors::StubNotFound.new('foo')
     => #<Excon::Error::StubNotFound: foo>

**Comment by geemus** [comment]
@starbelly seems alright. It wouldn't help with the `Foo` vs `FooError`, but I guess we could take a similar approach to those? ie also making `HTTPStatus` and duping that to have `HTTPStatusError`? Anyway, that was more what this issue was about (if I recall properly) than `Error `vs `Errors`.

**Comment by starbelly** [comment]
@geemus Yeah, realize that. Right now it's Excon::Errors::Error.new() and I suppose that's where I started with the idea of refactoring it out into Excon::Error. I  figured if it's going to be standardized, might as well refactor while in the process of s/FooError/Foo.  Hmmm, this would also need some more adjustment as we would want to inform the caller that Excon::Errors is deprecated.

**Comment by geemus** [comment]
Yeah, makes sense. Definitely worth discussing.

**Comment by starbelly** [comment]
@geemus - Perhaps more like this:

```ruby
module Excon
  module Error
    class Standard < StandardError
    end
    def self.new(msg)
      Excon::Error::Standard.new(msg)
    end
    class Foo < Error::Standard; end
  end

  # Legacy
  module Errors
    class Error < StandardError
      def self.new(msg)
        old_space = self.to_s
        old_space =~ /Excon::Errors::(\w+)/
        new_space = "Excon::Error::" + Regexp.last_match(1).sub('Error', '')
        warn "#{old_space} is deprecated, please use #{new_space}, I pitty da foo!"
        raise self, msg
      end
    end
    class FooError < Errors::Error; end
  end
end
```

 Then way later down the road, modify this to warn and throw the new error instead of continuing with the old.

**Comment by geemus** [comment]
Yeah, seems like something along those lines would get us there.

**Comment by starbelly** [comment]
@geemus I'll put a PR together and we can go from there

**Comment by geemus** [comment]
Sounds great, thanks!

**Comment by geemus** [comment]
closed by #571
